switchbot/devices/lock.py

```diff
@@ -86,77 +83,78 @@ async def verify_encryption_key(
 
         return lock_info is not None
 
+    @staticmethod
+    def api_request(subdomain: str, path: str, data: dict, headers: dict = None):
```

Comment:
Since the rest of this library is asyncio, I think this would be a good opportunity to switch to aiohttp instead as there isn't anything that specifically needs requests or a sync library anymore